frontend/apps/reader/modules/readerrolling.lua

```diff
@@ -783,12 +850,13 @@ function ReaderRolling:updatePos()
     local new_height = self.ui.document.info.doc_height
     local new_page = self.ui.document.info.number_of_pages
     if self.old_doc_height ~= new_height or self.old_page ~= new_page then
+        if self.hide_nonlinear_flows then
+            self.ui.document:cacheFlows()
+        end
         self:_gotoXPointer(self.xpointer)
         self.old_doc_height = new_height
         self.old_page = new_page
         self.ui:handleEvent(Event:new("UpdateToc"))
-        self.view.footer:setTocMarkers(true)
-        self.view.footer:onUpdateFooter()
```

Comment:
You removed these ones (and one below), because they are not needed, or triggered by something else?

Reply:
They are triggered in `ReaderFooter:onTocUpdated()`. I thought it's better if the footer manages its own updates, rather than rely on the reader asking for explicit updates.

Reply:
OK, agreed.